### PR TITLE
Select subset of rows

### LINKED ISSUE #2643: Select subset of rows
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Screenshot: Prototype
[![a screenshot of ...][image]][image-link]
<sup>[Click screenshot to visit Figma.][image-link]</sup>

## Acceptance Criteria
- [ ] Multiple rows can be selected
- [ ] Selected rows are stored in some local state, which can be referenced by other functionality.

   [image]: https://user-images.githubusercontent.com/305339/166574896-aaf72687-608e-412b-87f7-07808ea5964e.png
   [image-link]: https://www.figma.com/proto/XS4Ag6GWcgdq2dBlLzBkay/Admin-v3?node-id=465%3A8698&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=410%3A1984&hide-ui=1
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

### PR DESCRIPTION
**Pull request by JamesHuf**
Will resolve #2643.

Adds selection checkboxes and ability to tables if passed in a callback on selection change.
Test by ensuring selectedRows in UserTable properly updates with the proper data.

Note: I'm not certain if this is the best way of going about this ~~and I currently have a minor issue.~~

### CODE REVIEW

File: frontend/admin/src/js/components/Table/Table.tsx [hunk 1]
```diff
@@ -54,6 +56,47 @@ const IndeterminateCheckbox: React.FC<
...
+    if (ref.current) {
+      ref.current.indeterminate = indeterminate;
+    }
+  }, [ref, indeterminate]);
+
+  return <input type="checkbox" ref={ref} {...rest} />;
+};
+
+// TODO: Try to fix these types
+const HeaderCheckbox = ({ getToggleAllPageRowsSelectedProps }) => {
```
Comment (JamesHuf) [inline frontend/admin/src/js/components/Table/Table.tsx:74]: I can't find any good way of getting the type for 'getToggleAllPageRowsSelectedProps' here and 'rows' in RowCheckbox. I'd love to hear a good solution for this.

Comment (esizer) [reply frontend/admin/src/js/components/Table/Table.tsx:74]: There are some [types that may work in @types/react-table](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/42c09f2892ae6923649f3465b0cd4951aa6aad01/types/react-table/index.d.ts#L738).

**Comment by esizer** [comment]
Just a heads up, I believe @petertgiles is in the process of refactoring `frontend/admin/src/js/components/Table/Table.tsx` and `frontend/admin/src/js/components/user/UserTable.tsx` which may cause a few merge conflicts in the near future with either PR.

File: frontend/admin/src/js/components/Table/Table.tsx (hunk 1 above)
Comment (JamesHuf) [reply frontend/admin/src/js/components/Table/Table.tsx]: That helped. Thanks!

**Comment by JamesHuf** [comment]
@petertgiles It looks like this is likely to have some merge conflicts with #2646. Should we work merging this in now or wait and merge your changes in first?

**Comment by vd1992** [comment]
based off of

**Comment by JamesHuf** [comment]
Good call!
I'll set this issue as blocked by #3173.

**Pull request closed without merging**